### PR TITLE
DEPR: deprecate notebook argument in DataFrame.to_html()

### LINKED ISSUE #23820: DISC: deprecate notebook parameter in to_html()
should to_html(notebook=True) be part of the public facing API or called just from _repr_html_()?

### PR DESCRIPTION
**Pull request by simonjayhawkins**
- [ ] closes #23820
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by WillAyd** [comment]
Code looks pretty good. Just to clarify since I'm not super familiar with this one, does this serve any purpose from an end user perspective or is it pure cruft? 

Just wondering if we should document the alternatives to this if any

**Comment by simonjayhawkins** [comment]
> Code looks pretty good. Just to clarify since I'm not super familiar with this one, does this serve any purpose from an end user perspective or is it pure cruft?

I'm not sure of the history. Assuming that it was needed internally but added to a public method. So I don't see it serving any purpose to an end user.

> Just wondering if we should document the alternatives to this if any

The alternative is the `_repr_html_` method. As this is not part of the public API, i'm reluctant to suggest this as an alternative.

File: doc/source/whatsnew/v0.25.0.rst [hunk 1]
```diff
@@ -495,7 +495,7 @@ Other Deprecations
   Use the public attributes :attr:`~RangeIndex.start`, :attr:`~RangeIndex.stop` and :attr:`~RangeIndex.step` instead (:issue:`26581`).
 - The :meth:`Series.ftype`, :meth:`Series.ftypes` and :meth:`DataFrame.ftypes` methods are deprecated and will be removed in a future version.
   Instead, use :meth:`Series.dtype` and :meth:`DataFrame.dtypes` (:issue:`26705`).
-
+- The ``notebook`` argument of :meth:`DataFrame.to_html` is deprecated (:issue:`23820`).
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.0.rst]: mention the replacement

File: pandas/core/frame.py [hunk 2]
```diff
@@ -650,8 +650,11 @@ def _repr_html_(self):
             max_cols = get_option("display.max_columns")
             show_dimensions = get_option("display.show_dimensions")
 
-            return self.to_html(max_rows=max_rows, max_cols=max_cols,
-                                show_dimensions=show_dimensions, notebook=True)
+            formatter = fmt.DataFrameFormatter(
```
Comment (jreback) [inline pandas/core/frame.py]: why the change?

File: pandas/core/frame.py [hunk 3]
```diff
@@ -2158,6 +2162,9 @@ def to_html(self, buf=None, columns=None, col_space=None, header=True,
             Convert the characters <, >, and & to HTML-safe sequences.
         notebook : {True, False}, default False
             Whether the generated HTML is for IPython Notebook.
+
+            .. deprecated:: 0.25.0
```
Comment (jreback) [inline pandas/core/frame.py]: is there a replacement?

**Review by jreback — CHANGES_REQUESTED** [review]
add the deprecation here: https://github.com/pandas-dev/pandas/issues/6581

File: doc/source/whatsnew/v0.25.0.rst (hunk 1 above)
Comment (simonjayhawkins) [reply doc/source/whatsnew/v0.25.0.rst]: there would be no replacement from an end user perspective. see https://github.com/pandas-dev/pandas/pull/26815#issuecomment-501402253

File: pandas/core/frame.py (hunk 2 above)
Comment (simonjayhawkins) [reply pandas/core/frame.py]: `_repr_html_` just calls `to_html` with `notebook=True`, so need to avoid the FutureWarning in the notebook.

eventually i'd like to have `_repr_html_` instantiate `NotebookFormatter` directly and `to_html` instantiate `HTMLFormatter` directly. But at the moment neither are subclasses of `DataFrameFormatter`

**Comment by simonjayhawkins** [comment]
> add the deprecation here: #6581

done

File: pandas/core/frame.py (hunk 3 above)
Comment (simonjayhawkins) [reply pandas/core/frame.py]: there would be no replacement from an end user perspective. see https://github.com/pandas-dev/pandas/pull/26815#issuecomment-501402253

**Review by WillAyd — CHANGES_REQUESTED** [review]
minor comment otherwise lgtm

**Comment by WillAyd** [comment]
General question on this - are we adding deprecations to the 1.0.0 release or removing outright?

**Comment by TomAugspurger** [comment]
I thought adding deprecations. May be wrong though.

**Review by WillAyd — APPROVED** [review]
lgtm @TomAugspurger

**Comment by TomAugspurger** [comment]
This deprecation docs / warning should have some indication for what users should do instead. What do we recommend?

**Comment by simonjayhawkins** [comment]
> This deprecation docs / warning should have some indication for what users should do instead. What do we recommend?

there would be no replacement from an end user perspective. see https://github.com/pandas-dev/pandas/pull/26815#issuecomment-501402253

**Comment by TomAugspurger** [comment]
So you're saying it's not useful to users? How do you know that?

Why don't we document what `notebook=True` used to imply? Or, alternatively,
we could just say `_repr_html_` is part of the API for this purpose. It's a
well established
interface that we aren't going to remove.

**Comment by jreback** [comment]
@simonjayhawkins can you summarize status here

**Comment by simonjayhawkins** [comment]
there is not agreement that this should be deprecated. so will close and maybe reopen in the future if there is a stronger argument for deprecation.
